### PR TITLE
Remove test builders from validate_params_test.go

### PR DESCRIPTION
**Pull request by popcor255**
# Changes

* Using minimal internal types are simpler than full pipeline objects. 
* Removed duplicate tests
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

/area testing
/kind cleanup

#3178

### CODE REVIEW

File: pkg/reconciler/pipelinerun/resources/validate_params_test.go [hunk 1]
```diff
@@ -86,92 +76,106 @@ func TestValidateParamTypesMatching_Valid(t *testing.T) {
...
-		p    *v1beta1.Pipeline
-		pr   *v1beta1.PipelineRun
+
+	mockStringValue := *v1beta1.NewArrayOrString("mockStringValue")
+	mockArrayValue  := *v1beta1.NewArrayOrString("mockArrayValue", "mockArrayValue")
+
+	for _, tc := range []struct {
+		name          string
+		description	  string
+		pp            []v1beta1.ParamSpec
```
Comment (popcor255) [inline pkg/reconciler/pipelinerun/resources/validate_params_test.go:86]: just why...
:crying_cat_face:

File: pkg/reconciler/pipelinerun/resources/validate_params_test.go [hunk 2]
```diff
@@ -86,92 +76,106 @@ func TestValidateParamTypesMatching_Valid(t *testing.T) {
...
+		prp			  []v1beta1.Param
+		errorExpected bool
```
Comment (popcor255) [inline pkg/reconciler/pipelinerun/resources/validate_params_test.go:88]: just why...
:crying_cat_face:

**Review by vdemeester — CHANGES_REQUESTED** [review]
> Test names use underscores instead of blank spaces. Golang test convert blanks to underscores into the terminal anyway. Changing in the code base ahead of time makes it easier to find failed test in editors.

I am not really a fan of doing that change. The fact that `go` replace converts blank to something is an implementation detail.

> A description was added to each test case.

Not a fan of that one either, the `name` field should be sufficient *and* it's gonna be extra burden to maintain. Just like most comments, it might become incorrect with what the test actually does.

**Comment by ImJasonH** [comment]
> > Test names use underscores instead of blank spaces. Golang test convert blanks to underscores into the terminal anyway. Changing in the code base ahead of time makes it easier to find failed test in editors.
> 
> I am not really a fan of doing that change. The fact that `go` replace converts blank to something is an implementation detail.

+1, I've even been changing underscores to spaces in other changes 😆 

> Not a fan of that one either, the `name` field should be sufficient _and_ it's gonna be extra burden to maintain. Just like most comments, it might become incorrect with what the test actually does.

Agree here too, test names should be short and descriptive enough to not need a longer description. If you do need to describe something in more detail, that's fine, but I'd do it in a `// comment` and not in code that gets printed with the test. Having `name` and `description` is also inconsistent with other tests throughout the codebase, and I don't think we should update them all to match.

**Comment by ImJasonH** [comment]
/kind cleanup

Poking the kind label check 🤞

**Comment by popcor255** [comment]
@vdemeester @ImJasonH Thanks for the feedback! :smile:
I made the changes. You can take another

File: pkg/reconciler/pipelinerun/resources/validate_params_test.go [hunk 3]
```diff
@@ -19,159 +19,133 @@ package resources
...
 
 func TestValidateParamTypesMatching_Valid(t *testing.T) {
-	tcs := []struct {
-		name          string
-		p             *v1beta1.Pipeline
-		pr            *v1beta1.PipelineRun
-		errorExpected bool
+
+	mockStringValue := *v1beta1.NewArrayOrString("mockStringValue")
+	mockArrayValue := *v1beta1.NewArrayOrString("mockArrayValue", "mockArrayValue")
```
Comment (vdemeester) [inline pkg/reconciler/pipelinerun/resources/validate_params_test.go:29]: nit: technically, this is not a "mock" :stuck_out_tongue_closed_eyes: we could name it `stringValue` or anything.
But it's a nit, no need to fix it now :wink:

**Review by vdemeester — APPROVED** [review]
/meow
/hold cancel

File: pkg/reconciler/pipelinerun/resources/validate_params_test.go (hunk 3 above)
Comment (popcor255) [reply pkg/reconciler/pipelinerun/resources/validate_params_test.go]:
```bash
%s/mockStringValue/stringValue/g
%s/mockArrayValue/arrayValue/g
```

**Comment by popcor255** [comment]
poke @dlorenc

**Review by jerop — COMMENTED** [review]
/lgtm

**Pull request merged**